lunatrace/bsl/backend/src/config/index.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {
+  AwsConfig,
+  BackendBucketConfig,
+  GithubAppConfig,
+  HasuraConfig,
+  JwksConfig,
+  LogConfig,
+  QueueHandlerConfig,
+  RepositoryQueueConfig,
+  ServerConfig,
+  SqsQueueConfig,
+  StaticAnalysisConfig,
+  WebhookConfig,
+  WorkerBucketConfig,
+} from '../types/config';
+import { tryParseInt, validateBooleanString } from '../utils/parse';
+
+import { envVars } from './load-environment-vars';
+
+export { envVars } from './load-environment-vars';
+export const isProduction = envVars.NodeEnv === 'production';
+
+// The below loaders build configs for various parts of the app.
+// TODO: There is no reason in node to load configs like this in NodeJS, its just extra code and not real DI.
```

Comment:
The reason the configuration was written like this was to allow dependency injection to be possible. I do not think environment variables should be referenced directly from a global context.